proto/internal/temporal/server/api/matchingservice/v1/request_response.proto

```diff
@@ -203,22 +203,28 @@ message GetWorkerBuildIdOrderingResponse {
 
 message InvalidateTaskQueueMetadataRequest {
     string namespace_id = 1;
+    // The name of the task queue, which always refers to / is applied to a workflow task queue.
     string task_queue = 2;
-    // The task queue versioning data should be invalidated, and fresh data fetched from the root
-    // partition.
-    bool versioning_data = 3;
+    // The task queue versioning data should be invalidated and replaced with this data, if set.
+    temporal.server.api.persistence.v1.VersioningData versioning_data = 3;
 }
 message InvalidateTaskQueueMetadataResponse {}
 
 message GetTaskQueueMetadataRequest {
     string namespace_id = 1;
     string task_queue = 2;
     // If set nonempty, the requester wants the latest versioning data. The value must be the hash
-    // (using farm hash Fingerprint32) of the latest versioning data. If the requester has no data,
+    // (using farm hash Fingerprint64) of the latest versioning data. If the requester has no data,
     // it can use any invalid value (ex: [0]).
     // If the data is up to date, no value will be returned.
     bytes want_versioning_data_curhash = 3;
 }
 message GetTaskQueueMetadataResponse {
-    temporal.server.api.persistence.v1.VersioningData versioning_data = 1;
+    oneof versioning_data_resp {
```

Comment:
bleh.. I understand why you want to use oneof but the api in go is rather unpleasant. do you think it would be so bad to just have a `VersioningData` and a `bool` at the top level?

Reply:
I can't bring myself to make the wrong model just b/c the Go codegen is nasty.